analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,503 @@
+
+```{r}
+# Input
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+pal_dir <- file.path(root_dir, "figures", "palettes")
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+pal_file <- file.path(pal_dir, "broad_histology_cancer_group_palette.tsv")
+
+# Output
+results_dir <- file.path("results")
+if (!(dir.exists(results_dir))) {
+  dir.create(results_dir)
+}
+rna_stranded_df_file <- file.path(results_dir, "metadata_rna_stranded_same-extraction.tsv")
+rna_stranded_thresholded_df_file <- file.path(results_dir, "metadata_thresholded_rna_stranded_same-extraction.tsv")
+```
+
+
+Read in:
+
+```{r}
+metadata <- read_tsv(metadata_file) 
+palette_mapping_df <- read_tsv(pal_file)
+```
+
+## Exploration of `tumor_fraction`
+
+The quantity `tumor_fraction` provides information about what percent of the sample was likely tumor. 
+We'll explore this variable:
+
+```{r}
+# Note that this quantity is only recorded, if ever, with WGS libraries: 
+presence_table <- table(metadata$experimental_strategy,
+                        is.na(metadata$tumor_fraction))
+colnames(presence_table) <- c("Tumor fraction known", "Tumor fraction unknown")
+presence_table
+```
+
+Tumor fraction was calculated for WGS samples:
+
+```{r}
+metadata_tumor_wgs <- metadata %>%
+  filter(experimental_strategy == "WGS", 
+         sample_type == "Tumor")
+
+# For what percentage of tumor samples is purity known? Nearly all of them!
+sum(!is.na(metadata_tumor_wgs$tumor_fraction)) / nrow(metadata_tumor_wgs)
+```
+
+
+Distribution of tumor purity:
+
+```{r}
+summary(metadata_tumor_wgs$tumor_fraction)
+
+ggplot(metadata_tumor_wgs) + 
+  aes(x = tumor_fraction) + 
+  geom_histogram(color = "black", fill = "skyblue")
+```
+
+Distribution by cancer group and broad histology:
+
+```{r}
+# combine metadata with palette:
+tumor_palette_df <- metadata_tumor_wgs %>%
+  inner_join(palette_mapping_df) %>%
+  select(Kids_First_Biospecimen_ID, 
+         tumor_fraction, 
+         cancer_group_display, 
+         cancer_group_hex, 
+         broad_histology_display,
+         broad_histology_order,
+         broad_histology_hex) %>%
+  drop_na() %>%
+  # reorder by median fraction and ensure "Other" is always last
+  mutate(cancer_group_display = forcats::fct_reorder(cancer_group_display, tumor_fraction), 
+         cancer_group_display = forcats::fct_relevel(cancer_group_display, "Other", after = Inf),
+         broad_histology_display = forcats::fct_reorder(broad_histology_display, tumor_fraction), 
+         broad_histology_display = forcats::fct_relevel(broad_histology_display, "Other", after = Inf))
+         
+# plotting function:
+tumor_violins <- function(df, display_name, display_hex) {
+  ggplot(df) + 
+    # order by ascending median
+    aes(x = forcats::fct_reorder({{display_name}}, tumor_fraction),
+        y = tumor_fraction, 
+        fill = {{display_hex}}) + 
+    geom_violin(alpha = 0.5) +
+    stat_summary() + 
```

Comment:
Is the default point for `stat_summary()` the median? If so, the two next plots don't look as I would I expect if ordering by median.

Reply:
The default is `mean_se()` but I can set it to use median now that the plot got reordered 👍